account_invoice_product_supplier_price_update/account_invoice.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.model
+    def _check_product_supplierprice(self):
+        lines = []
+        for line in self.invoice_line:
+            current_price_unit = [(6, 0, [])]
```

Comment:
The code can be more simple here.
- first the current_price_unit do not need the 6, 0 as you always use the [0][2] after so you can remove it.
- then you do not need all of the variable suppinfo...

`
something like that:

    def _check_product_supplierprice(self):                         
    lines = []                                                             
    for line in self.invoice_line:                                         
        current_price_unit = []
        supplier = self.partner_id.commercial_partner_id or self.partner_id
	suppinfo = False
        current_price_uni = []
        for seller in line.product_id.seller_ids:                          
            if supplier == seller.name: 
                suppinfo = seller
                current_price_unit = [pricelist.price for p in suppinfo.pricelist_ids]
                break                                                      
        if line.price_unit not in current_price_unit:                                               
            lines.append((0, 0, {                                          
                'name': line.name,                                         
                'current_price_unit': suppinfo.unit_price_note,            
                'new_price_unit': line.price_unit,                         
                'suppinfo_id': suppinfo.id,                                
                'to_variant': True,                                        
                'supplier_id': supplier_id.id,                             
                'product_id': line.product_id.id,                          
                'product_tmpl_id': line.product_id.product_tmpl_id.id,     
            }))                                                            
    return lines                                                           
`